### PR TITLE
Add a NumberQuestionDefinition

### LINKED ISSUE #392: "Number" question type: implement bare bones
This question type produces a numerical input widget with an optional spinner.

image

Make it possible for UAT admins to create, and for applicants to fill out, questions of this type. Answers should be saved to a database. Configuration options and validation criteria aren't necessary yet.

See more background info at https://github.com/seattle-uat/universal-application-tool/wiki/Question-type-requirements-notes

### PR DESCRIPTION
**Pull request by carolinedanzi**
### Description
1. Add a new question type for whole numbers: `NumberQuestionDefinition`
2. Implement rendering for number questions on the applicant side
3. Allow admins to configure min/max value allowed for number questions

### Checklist
- [x] Created tests which fail without the change (if possible)
- [x] Extended the README / documentation, if necessary

### Issue(s)
Fixes #392

Screen Shot 2021-03-22 at 4 29 07 PM
Screen Shot 2021-03-22 at 4 08 33 PM
Screen Shot 2021-03-22 at 4 05 53 PM

### CODE REVIEW

File: universal-application-tool-0.0.1/app/services/question/NumberQuestionDefinition.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  }
+
+  @Override
+  public QuestionType getQuestionType() {
+    return QuestionType.NUMBER;
+  }
+
+  @Override
+  public ImmutableMap<Path, ScalarType> getScalars() {
+    return ImmutableMap.of(getNumberPath(), getNumberType());
```
Comment (hmhsu) [inline universal-application-tool-0.0.1/app/services/question/NumberQuestionDefinition.java:117]: Question(blocking): Do we need to include the metadata?

File: universal-application-tool-0.0.1/app/views/admin/questions/QuestionEditView.java [hunk 2]
```diff
@@ -121,7 +121,7 @@ private ContainerTag buildQuestionForm(QuestionForm questionForm) {
                 .setFieldName("questionHelpText")
                 .setLabelText("Question help text")
                 .setPlaceholderText("The question help text displayed to the applicant")
-                .setValue(questionForm.getQuestionText())
+                .setValue(questionForm.getQuestionHelpText())
```
Comment (hmhsu) [inline universal-application-tool-0.0.1/app/views/admin/questions/QuestionEditView.java:124]: Praise: Thanks for catching this!

File: universal-application-tool-0.0.1/app/services/question/NumberQuestionDefinition.java (hunk 1 above)
Comment (carolinedanzi) [reply universal-application-tool-0.0.1/app/services/question/NumberQuestionDefinition.java]: Oof! Great catch - I wonder if we can determine a way to make this less forgettable

File: universal-application-tool-0.0.1/app/views/questiontypes/NumberQuestionRenderer.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  public NumberQuestionRenderer(ApplicantQuestion question) {
+    this.question = checkNotNull(question);
+  }
+
+  @Override
+  public Tag render() {
+    ApplicantQuestion.NumberQuestion numberQuestion = question.getNumberQuestion();
+
+    return label(
+        text(question.getQuestionText()),
```
Comment (hmhsu) [inline universal-application-tool-0.0.1/app/views/questiontypes/NumberQuestionRenderer.java:25]: Thoughts(non-blocking): Aaron has a PR that enhances existing question types, might worthing checkout out there (and maybe matching the styles).

File: universal-application-tool-0.0.1/test/services/applicant/ApplicantQuestionTest.java [hunk 4]
```diff
@@ -99,6 +108,54 @@ public void textQuestion_withPresentApplicantData_failsValidation() throws Excep
     assertThat(textQuestion.getTextValue().get()).isEqualTo("hello");
   }
 
+  @Test
```
Comment (hmhsu) [inline universal-application-tool-0.0.1/test/services/applicant/ApplicantQuestionTest.java:111]: Praise: thanks for adding the tests!

**Review by hmhsu — APPROVED** [review]
Looks great! Also thanks for adding the wiki page for adding a new question type.

File: universal-application-tool-0.0.1/app/services/question/NumberQuestionDefinition.java (hunk 1 above)
Comment (hmhsu) [reply universal-application-tool-0.0.1/app/services/question/NumberQuestionDefinition.java]: I could image having a protected method for those fields and call the method here? Alternatively, we could have a public base method that call all overridden methods (with a different name and make them protected) instead.

File: universal-application-tool-0.0.1/app/views/questiontypes/NumberQuestionRenderer.java (hunk 3 above)
Comment (carolinedanzi) [reply universal-application-tool-0.0.1/app/views/questiontypes/NumberQuestionRenderer.java:25]: Awesome, I hadn't seen that yet! I'll check it out

Comment (carolinedanzi) [reply universal-application-tool-0.0.1/app/views/questiontypes/NumberQuestionRenderer.java]: Updated to match `TextQuestionRenderer` - thanks for the heads up!

**Pull request merged**